test/utils/helper.js

```diff
@@ -1380,6 +1380,98 @@ const transferRewardSafe = async function (
   );
 };
 
+const withdrawFromRewardSafe = async function (
+  rewardManager,
+  rewardSafe,
+  tokenAddress,
+  value,
+  relayer,
+  gasToken
+) {
+  const rewardSafeEOA = (await rewardSafe.getOwners())[1];
+
+  const transferData = AbiCoder.encodeFunctionCall(
+    {
+      name: "transfer",
+      type: "function",
+      inputs: [
+        {
+          type: "address",
+          name: "to",
+        },
+        {
+          type: "uint256",
+          name: "value",
+        },
+      ],
+    },
+    [rewardSafeEOA, value]
+  );
+
+  const nonce = await rewardSafe.nonce();
+
+  const fullSignatureInnerExec = await rewardEIP1271Signature(
+    rewardSafe.address,
+    0,
+    transferData,
+    0,
+    0,
+    0,
+    0,
+    gasToken.address,
+    rewardSafe.address,
+    nonce.add(toBN("1")),
+    rewardSafeEOA,
+    rewardSafe,
+    rewardManager
+  );
+
+  let payload = rewardManager.contract.methods
+    .withdrawFromRewardSafe(
+      tokenAddress,
+      value,
+      0,
+      0,
+      0,
+      gasToken.address,
+      fullSignatureInnerExec
+    )
+    .encodeABI();
+
+  const fullSignature = await rewardEIP1271Signature(
+    rewardManager.address,
+    0,
+    payload,
+    0,
+    0,
+    0,
+    0,
+    gasToken.address,
+    rewardSafe.address,
+    nonce,
+    rewardSafeEOA,
+    rewardSafe,
+    rewardManager
+  );
+
+  let safeTxData = {
+    to: rewardManager.address,
+    data: payload,
+    operation: 0,
+    txGasEstimate: 0,
+    gasPrice: 0,
```

Comment:
Perhaps, try adding `gasPrice`.  Since there are funds transferred we can collect gas -- look at `claimReward`. I don't do this for transferring ownership of safe because there may not be funds in the safe when it is transferred.

Reply:
You can check the transfer by replicating this  [test](https://github.com/cardstack/card-pay-protocol/blob/13dbac1f5149004086818f01843d344bfc94c910/test/RewardPool-test.js#L672)

Reply:
@tintinthong i don't think it's worth while to change this from 0--that is all out of the box gnosis safe contract logic. It is the GnosisSafe.sol contract that actually deducts the gas fee out of the safe. That is not code we have control over--moreover gnosis has already written tests for it. I think it is only worthwhile to create test around gas price when we are actually doing something with it (like the prepaid card manager's gas policies). I would say leave this at 0 to make our lives simpler and rest assured on the fg\act that gnosis is testing their own contract to be able to deduct gas fees from the safes appropriately.

Reply:
ok that is fine by me. I feel for me the value in doing this test is communication to someone who hasn't used gnosis stuff before. Honestly, when I first used these I would always wonder "why there is no gas fees paid out or why the gasPrice was 0". But if we just want our test to focus on the business logic and not be obscured by these gas payments -- I can see the benefit too.

Reply:
> why there is no gas fees paid out or why the gasPrice was 0"

Heh yeah this was me. Also I had thought that changing the gas price / adding the estimate was the first thing that made it work at all, but I might be wrong there, I'm continuing to dig